src/deploy/functions/params.ts

```diff
@@ -0,0 +1,339 @@
+import { logger } from "../../logger";
+import { FirebaseError } from "../../error";
+import { promptOnce } from "../../prompt";
+import * as build from "./build";
+import { assertExhaustive } from "../../functional";
+
+type CEL = build.Expression<string> | build.Expression<number> | build.Expression<boolean>;
+
+function isCEL(expr: string | number | boolean): expr is CEL {
+  return typeof expr === "string" && expr.includes("{{") && expr.includes("}}");
+}
+
+function dependenciesCEL(expr: CEL): string[] {
+  const deps: string[] = [];
+  const paramCapture = /{{ params\.(\w+) }}/g;
+  let match: RegExpMatchArray | null;
+  while ((match = paramCapture.exec(expr)) != null) {
+    deps.push(match[1]);
+  }
+  // return /{{ params\.\w+ }}/.exec(expr)?.slice(1) || [];
+  return deps;
+}
+
+/**
+ * Resolves a numeric field in a Build to an an actual numeric value.
+ * Fields can be literal or a {{ }} delimited CEL expression referencing param values.
+ * Currently only the CEL identity {{ params.PARAMNAME }} is implemented.
+ */
+export function resolveInt(
+  from: number | build.Expression<number>,
+  paramValues: Record<string, build.Field<string | number | boolean>>
+): number {
+  if (typeof from === "number") {
+    return from;
+  }
+  const match = /\A{{ params\.(\w+) }}\z/.exec(from);
+  if (!match) {
+    throw new FirebaseError("CEL evaluation of expression '" + from + "' not yet supported");
```

Comment:
prefer template strings.

```suggestion
    throw new FirebaseError(`CEL evaluation of expression "${from}" not yet supported`);
```